using_images/s2i_images/ruby.adoc

```diff
@@ -117,6 +124,40 @@ link:../../dev_guide/environment_variables.html#set-environment-variables[the
 $ oc env dc/rails-app RAILS_ENV=development
 ----
 
+== Performance tuning
+You can tune the number of threads per worker using the
+`*PUMA_MIN_THREADS*` and `*PUMA_MAX_THREADS*` environment variables.
+Additionally, the number of worker processes is determined by the number of CPU
+cores that the container has available, as recommended by
+link:https://github.com/puma/puma[Puma]'s documentation. This is determined using
+the cgroup link:https://www.kernel.org/doc/Documentation/cgroup-v1/cpusets.txt[cpusets]
+subsystem. You can specify the cores that the container is allowed to use by passing
+the `--cpuset-cpus` parameter to the link:http://docker.io[Docker] run command:
+
+----
+$ docker run -e PUMA_MAX_THREADS=32 --cpuset-cpus='0-2,3,5' -p 8080:8080 sinatra-app
+----
+
+The number of workers is also limited by the memory limit that is enforced using
+cgroups. The builder image assumes that you will need 50 MiB as a base and
+another 15 MiB for every worker process plus 128 KiB for each thread. Note that
+each worker has its own threads, so the total memory required for the whole
+container is computed using the following formula:
+
+----
+50 + 15 * WORKERS + 0.125 * WORKERS * PUMA_MAX_THREADS
+----
+
+You can specify a memory limit using the `--memory` flag:
+
+----
+$ docker run -e PUMA_MAX_THREADS=32 --memory=300m -p 8080:8080 sinatra-app
```

Comment:
same thing here.

we should be talking about resource limits/quota, not docker command args